contexts/MembershipContext/src/UseCases/ApplyForMembership/ApplyForMembershipUseCase.php

```diff
@@ -56,6 +60,11 @@ public function applyForMembership( ApplyForMembershipRequest $request ): ApplyF
 		if ( $this->policyValidator->isAutoDeleted( $application ) ) {
 			$application->markAsDeleted();
 		}
+
+		if ( $application->getPayment()->getPaymentMethod()->getType() === PaymentType::PAYPAL ) {
+			$application->setFirstPaymentDate( $this->paymentDelayCalculator->calculateFirstPaymentDate()->format( 'Y-m-d' ) );
+		}
```

Comment:
This strikes me as a tell-dont-ask violation. Also not good that this makes the UC bind to a specific payment type. My intuitive reaction here is that this should go into the domain, much like we did when we refactored some stuff in AddDonationUseCase into Donation::notifyOfPolicyValidationFailure. I need to look at this more to figure out what this would look like though.